tests/apps/core/test_core_models_meetings_get_occurences.py

```diff
@@ -0,0 +1,602 @@
+"""Unit tests for the `get_occurrences` method on the Meeting model."""
+import random
+from datetime import date
+
+from django.core.exceptions import ValidationError
+from django.test import TestCase
+
+from magnify.apps.core.factories import MeetingFactory
+
+
+class OccurencesMeetingsModelsTestCase(TestCase):
+    """Unit test suite to validate the behavior of meeting occurrences."""
+
+    # No recurence
+
+    def test_models_meetings_get_occurrences_no_recurrence(self):
+        """A meeting without recurrence should return 1 occurence."""
+        meeting = MeetingFactory(
+            start=date(2022, 7, 7),
+            recurrence=None,
+        )
+        self.assertEqual(
+            meeting.get_occurrences(date(2020, 3, 15), date(2050, 1, 1)),
+            [date(2022, 7, 7)],
+        )
+        self.assertEqual(
+            meeting.get_occurrences(date(2022, 7, 8), date(2050, 1, 1)),
+            [],
+        )
+
+    # Daily
+
+    def test_models_meetings_get_occurrences_daily_nb_occurrences_filled(self):
+        """
+        Daily occurences with date of end of reccurrence filled in but not number of occurrences.
+        """
+        meeting = MeetingFactory(
+            start=date(2022, 7, 7),
+            recurrence="daily",
+            frequency=1,
+            until=date(2022, 8, 2),
+            nb_occurrences=None,
+        )
+        self.assertEqual(meeting.nb_occurrences, 27)
+        self.assertEqual(
+            meeting.get_occurrences(date(2020, 3, 15), date(2050, 1, 1)),
+            [date(2022, 7, i) for i in range(7, 32)]
+            + [date(2022, 8, 1), date(2022, 8, 2)],
+        )
+
+        meeting = MeetingFactory(
+            start=date(2022, 7, 7),
+            recurrence="daily",
+            frequency=3,
+            until=date(2022, 8, 2),
+            nb_occurrences=None,
+        )
+        self.assertEqual(meeting.nb_occurrences, 9)
+        # The date of end of recurrence should be corrected
+        self.assertEqual(meeting.until, date(2022, 7, 31))
+        self.assertEqual(
+            meeting.get_occurrences(date(2020, 3, 15), date(2050, 1, 1)),
+            [date(2022, 7, 7 + 3 * i) for i in range(9)],
```

Comment:
I find this assert hard to read…
IMHO, it would be clearer with a simple hardcoded dates list.

Reply:
:+1: